cmd/producer/main_test.go

```diff
@@ -27,13 +27,55 @@ type fakeRedis struct {
 	client redis.Cmdable
 }
 
+func TestRedisClientSetup(t *testing.T) {
+	tests := []struct {
+		name			string
+		cert			string
+	}{{
+		name:			"test with fake cert from x509",
+		cert:			"-----BEGIN CERTIFICATE-----\n" +
+						"MIIDBjCCAe6gAwIBAgIRANXM5I3gjuqDfTp/PYrs+u8wDQYJKoZIhvcNAQELBQAw\n" +
+						"EjEQMA4GA1UEChMHQWNtZSBDbzAeFw0xODAzMjcxOTU2MjFaFw0xOTAzMjcxOTU2\n" +
+						"MjFaMBIxEDAOBgNVBAoTB0FjbWUgQ28wggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAw\n" +
+						"ggEKAoIBAQDK+9m3rjsO2Djes6bIYQZ3eV29JF09ZrjOrEHLtaKrD6/acsoSoTsf\n" +
+						"cQr+rzzztdB5ijWXCS64zo/0OiqBeZUNZ67jVdToa9qW5UYe2H0Y+ZNdfA5GYMFD\n" +
+						"yk/l3/uBu3suTZPfXiW2TjEi27Q8ruNUIZ54DpTcs6y2rBRFzadPWwn/VQMlvRXM\n" +
+						"jrzl8Y08dgnYmaAHprxVzwMXcQ/Brol+v9GvjaH1DooHqkn8O178wsPQNhdtvN01\n" +
+						"IXL46cYdcUwWrE/GX5u+9DaSi+0KWxAPQ+NVD5qUI0CKl4714yGGh7feXMjJdHgl\n" +
+						"VG4QJZlJvC4FsURgCHJT6uHGIelnSwhbAgMBAAGjVzBVMA4GA1UdDwEB/wQEAwIF\n" +
+						"oDATBgNVHSUEDDAKBggrBgEFBQcDATAMBgNVHRMBAf8EAjAAMCAGA1UdEQQZMBeC\n" +
+						"FVRlc3RTeXN0ZW1DZXJ0UG9vbC5nbzANBgkqhkiG9w0BAQsFAAOCAQEAwuSRx/VR\n" +
+						"BKh2ICxZjL6jBwk/7UlU1XKbhQD96RqkidDNGEc6eLZ90Z5XXTurEsXqdm5jQYPs\n" +
+						"1cdcSW+fOSMl7MfW9e5tM66FaIPZl9rKZ1r7GkOfgn93xdLAWe8XHd19xRfDreub\n" +
+						"YC8DVqgLASOEYFupVSl76ktPfxkU5KCvmUf3P2PrRybk1qLGFytGxfyice2gHSNI\n" +
+						"gify3K/+H/7wCkyFW4xYvzl7WW4mXxoqPRPjQt1J423DhnnQ4G1P8V/vhUpXNXOq\n" +
+						"N9IEPnWuihC09cyx/WMQIUlWnaQLHdfpPS04Iez3yy2PdfXJzwfPrja7rNE+skK6\n" +
+						"pa/O1nF0AfWOpw==\n" +
+						"-----END CERTIFICATE-----\n",
+	}, {
+		name:			"test with empty cert",
+		cert:			"-----BEGIN CERTIFICATE-----" +
+						"-----END CERTIFICATE-----",
+	}}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			env = envInfo{
+				StreamName:			"mystream",
+				RedisAddress:		"rediss://redis.redis.svc.cluster.local:6379",
+				TlsCert:			test.cert,
+			}
+			setUpRedis()
+		})
+	}
+}
+
 func TestHandleRequest(t *testing.T) {
 	testserver := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		if r.Method != http.MethodGet && r.Method != http.MethodPost {
 			t.Errorf("Expected 'POST' OR 'GET' request, got '%s'", r.Method)
 		}
 	}))
-	setupRedis()
+	setupFakeRedis()
```

Comment:
Why are we setting up “fake” Redis if this is an e2e test?

Typically fake things are only used for integration and unit tests not e2e tests — which by definition need be end to end

Reply:
This is a unit test, there was a name duplication when moving the [redis client out of the main function](https://github.com/knative-sandbox/async-component/blob/177dac1194cdab55969ac3b01ba7e7c3dfb0d05c/cmd/producer/main.go#L81) in producer/main.go  

I thought it prudent to point out that this is a dummy one, which was already used there in that capacity, while also removing the amount of logic that lived in the main function of the producer.